File: apps/mobile/ios/RoamPulseWidget/Shared/WidgetPersistedData.swift
import Foundation

/// The raw fields persisted by `WidgetDataStore` from Flutter's
/// MethodChannel calls — one group written independently per call from
/// ConnectivityBloc's and DashboardBloc's own updates, matching the
/// Dart-side `ConnectivityWidgetData`/`PlanWidgetData` split and
/// Android's `WidgetPersistedData`.
struct WidgetPersistedData {
    let connectionState: String?
    let carrierName: String?
    let technology: String?
    let connectivityLastSyncedAt: String?
    let destinationCity: String?
    let countryCode: String?
    let dataRemainingMb: Double?
    let dataAllowanceMb: Double?
    let daysRemaining: Int?
}

File: apps/mobile/android/app/src/main/kotlin/com/example/roam_pulse/widget/WidgetPersistedData.kt
package com.example.roam_pulse.widget

/**
 * The raw fields persisted by [WidgetDataRepository] from Flutter's
 * MethodChannel calls — one group written independently per call from
 * ConnectivityBloc's and DashboardBloc's own updates, matching the
 * Dart-side `ConnectivityWidgetData`/`PlanWidgetData` split.
 */
data class WidgetPersistedData(
    val connectionState: String?,
    val carrierName: String?,
    val technology: String?,
    val connectivityLastSyncedAt: String?,
    val destinationCity: String?,
    val countryCode: String?,
    val dataRemainingMb: Double?,
    val dataAllowanceMb: Double?,
    val daysRemaining: Int?,
)
